bridge-impl/src/main/java/com/liferay/faces/bridge/renderkit/html_basic/internal/BodyRendererBridgeImpl.java

```diff
@@ -190,20 +189,22 @@ public void encodeEnd(FacesContext facesContext, UIComponent uiComponent) throws
 
 		// If non-Ajax request, render scripts from FacesRequestContext.
 		if (!facesContext.getPartialViewContext().isAjaxRequest()) {
-
-			encodeScripts(facesContext, responseWriter, uiComponent);
+			encodeScripts(responseWriter);
 		}
 
 		// Render the closing </div> tag.
 		responseWriter.endElement(ELEMENT_DIV);
 	}
 
-	protected void encodeScripts(FacesContext facesContext, ResponseWriter responseWriter, UIComponent uiComponent)
-		throws IOException {
+	protected void encodeScripts(ResponseWriter responseWriter) throws IOException {
+
+		responseWriter.startElement("script", null);
+		responseWriter.writeAttribute("type", "text/javascript", null);
 
-		BodyScriptEncoderFactory bodyScriptEncoderFactory = (BodyScriptEncoderFactory) BridgeFactoryFinder.getFactory(
-				BodyScriptEncoderFactory.class);
-		BodyScriptEncoder bodyScriptEncoder = bodyScriptEncoderFactory.getBodyScriptEncoder();
-		bodyScriptEncoder.encodeScripts(facesContext, responseWriter, uiComponent);
+		ScriptEncoderFactory scriptEncoderFactory = (ScriptEncoderFactory) FactoryExtensionFinder.getFactory(
+				ScriptEncoderFactory.class);
+		ScriptEncoder scriptEncoder = scriptEncoderFactory.getScriptEncoder();
+		scriptEncoder.encodeScripts(responseWriter);
+		responseWriter.endElement("script");
```

Comment:
This code may need slight refactoring because it leaves behind an empty `<script>` when in Liferay since scripts are being buffered to the bottom of the page. However, since this code may need to be refactored according to [this comment](https://github.com/liferay/liferay-faces-util/pull/6#diff-for-comment-43178297) anyway, I'm gonna wait til we decide what to do with that refactor before fixing this minor issue.